arangod/Sharding/ShardingStrategyDefault.cpp

```diff
@@ -369,12 +372,14 @@ ShardingStrategyEnterpriseBase::ShardingStrategyEnterpriseBase(
   TRI_ASSERT(!shardKeys.empty());
 
   if (shardKeys.size() == 1) {
+    TRI_ASSERT(!shardKeys[0].empty());
+
     _usesDefaultShardKeys =
         (shardKeys[0] == StaticStrings::KeyString ||
-         (shardKeys[0][0] == ':' &&
+         (shardKeys[0].starts_with(':') &&
           shardKeys[0].compare(1, shardKeys[0].size() - 1,
                                StaticStrings::KeyString) == 0) ||
-         (shardKeys[0].back() == ':' &&
+         (shardKeys[0].ends_with(':') &&
           shardKeys[0].compare(0, shardKeys[0].size() - 1,
```

Comment:
starts_with?

Reply:
I think substituting the "compare" part with "starts_with" wouldn't be exactly identical.
It is true that using "starts_with" instead of compare would check if the string starts with `StaticStrings::KeyString`, but this would be also be true for strings with arbitrary other characters in between.

For example:
**Good**
shardKeys[0]: "_key:"
ends_with(':'): true
compare(0, shardsKeys[0].size() - 1, StaticStrings::KeyString): 0
starts_with(StaticStrings::KeyString): true

**Bad**
shardKeys[0]: "_keyabcdef:"
ends_with(':'): true
compare(0, shardsKeys[0].size() - 1, StaticStrings::KeyString): 1
starts_with(StaticStrings::KeyString): true

Replacing the "compare" functions with "starts_with"/"ends_with" thus would require an extra check for the string length, which is fine to add, but wasn't done by this PR.

Reply:
You're right. Btw maybe better constant and ==?
something like
```cpp
static constexpr std::string_view kShardKey = ":key";
static constexpr std::string_view kKey = kShardKey.substr(1);
```

Reply:
I just found that we already have some StaticString constants for `:_key` and `_key:`, so I am now using these instead.

Reply:
See https://github.com/arangodb/arangodb/pull/20322/commits/56661b03871d22eb11c6d096ba31c4605c4c683a